stubs/usb.pyi

```diff
@@ -0,0 +1,37 @@
+"""
+Type stubs for PyUSB.
+
+Note that stubs are only written for the parts that we use.
+"""
+
+from typing import Generator, Optional
+
+class Configuration:
+    def ___init__(self, device: Device, configuration: int = 0) -> None: ...
```

Comment:
There's a lot of definitions in this file that aren't used (as evidenced by the fact that the name of this function contains five underscores and yet it hasn't caused any typechecking errors). I would recommend removing them, so that we're not maintaining things which are not used.

Specifically, I think the following can be removed from this type stub file:

- The `Configuration` class
- `Device.product` and `Device.manufacturer`
- `Device.set_configuration` and `Device.get_active_configuration`
- `Device.reset`

Yes, some of these are useful and we might use them some day. We're not using them right now though, and its easy enough to add solely the required ones back in if/when it turns out we do need them.